### ISSUE TITLE
Using a config file

### ISSUE THREAD

**Issue by gege251** [mlabs-haskell/bot-plutus-interface#60]
I think our PABConfig got a bit bloated with a lot of options that don't necessarily need to be changed.
I would propose a yaml configuration file, but i am open to other options.

**Comment by samuelWilliams99** [comment]
I quite like this package
https://hackage.haskell.org/package/config-value-0.8.2/docs/Config.html

**Comment by samuelWilliams99** [comment]
We should keep both options available though, as sometimes the config needs to be derived from something. If we expose a method like `getConfig :: FilePath -> Either Text PABConfig`, we can allow modifications to the config

**Comment by samuelWilliams99** [comment]
Didn't meant to click close there!

**Comment by ryukzak** [comment]
Config file format options (mostly from my experience):
- `yaml` -- good for programmers, but horrible for normal people who can use ms notebook, can't distinguish spaces and tabs, don't believe in space meaning.
- `dhall` -- high customizable configuration files with embedded template language, as I can see popular in Haskell community, but is overkill for our task and hard to use by normal people.
- `edn` -- default config format for clojure, looks good and easy to edit but has too unusual syntax for normal people/admins.
- A custom format (including what Sam mentioned above) can raise tooling problems. Usage in non-Haskell projects requires parser rewriting, which is not familiar for unix admins.
- `toml` -- extension for `ini` files, so it is a simple and familiar syntax for users.

In conclusion, I would like to try toml, but it is tricky to work with ADT. So I will make a few experiments.

**Comment by mikekeke** [comment]
Closed by #91
